packages/tiptap-commands/src/commands/markInputRule.js

```diff
@@ -3,23 +3,28 @@ import { InputRule } from 'prosemirror-inputrules'
 export default function (regexp, markType, getAttrs) {
   return new InputRule(regexp, (state, match, start, end) => {
     const attrs = getAttrs instanceof Function ? getAttrs(match) : getAttrs
-        const { tr } = state
-        let markEnd = end
+    const { tr } = state
+    const m = match.length - 1
+    let markEnd = end
+    let markStart = start
 
-    if (match[1]) {
-      const startSpaces = match[0].search(/\S/)
-      const textStart = start + match[0].indexOf(match[1])
-      const textEnd = textStart + match[1].length
-      if (textEnd < end) {
-        tr.delete(textEnd, end)
+    if (match[m]) {
+      const matchStart = start + match[0].indexOf(match[m - 1])
+      const matchEnd = matchStart + match[m - 1].length
```

Comment:
@philippkuehn actually the issue is here. but i don't get why... because if you add -1 (which i actually had before) when deleting the prefix it is range 4,6 (three positions) and when deleting the suffix it is 8,9 (just two positions) which is weird in my eyes but maybe you can come up   with an explanation i will just force push the -1 again which also had no side effects for my tests....

Reply:
okay... the match seems to be handled before the last input character actually reaches the transaction thats why there is this index weirdness at the end of the input...